pkg/sdk/sweepers_test.go

```diff
@@ -37,90 +39,170 @@ func TestSweepAll(t *testing.T) {
+		for _, c := range allClients {
+			err := nukeDatabases(c, "")()
+			assert.NoError(t, err)
+		}
+	})
 
-		err = nukeDatabases(secondaryClient, "acc_test_db_%")()
-		assert.NoError(t, err)
+	// TODO [SNOW-955520]:
+	t.Run("sweep warehouses", func(t *testing.T) {
+		t.Skipf("Used for manual sweeping; will be addressed during SNOW-955520")
+		for _, c := range allClients {
+			err := nukeWarehouses(c, "")()
+			assert.NoError(t, err)
+		}
 	})
 
 	// TODO [SNOW-955520]: nuke stale objects (e.g. created more than 2 weeks ago)
 }
 
 // TODO [SNOW-955520]: generalize nuke methods (sweepers too)
 func nukeWarehouses(client *Client, prefix string) func() error {
+	protectedWarehouses := []string{
+		"SNOWFLAKE",
+		"SYSTEM$STREAMLIT_NOTEBOOK_WH",
+	}
+
 	return func() error {
 		log.Printf("[DEBUG] Nuking warehouses with prefix %s\n", prefix)
 		ctx := context.Background()
 
-		whs, err := client.Warehouses.Show(ctx, &ShowWarehouseOptions{Like: &Like{Pattern: String(prefix)}})
+		var like *Like = nil
+		if prefix != "" {
+			like = &Like{Pattern: String(prefix)}
+		}
+
+		whs, err := client.Warehouses.Show(ctx, &ShowWarehouseOptions{Like: like})
 		if err != nil {
 			return fmt.Errorf("sweeping warehouses ended with error, err = %w", err)
 		}
+		var errs []error
 		log.Printf("[DEBUG] Found %d warehouses matching search criteria\n", len(whs))
 		for idx, wh := range whs {
 			log.Printf("[DEBUG] Processing warehouse [%d/%d]: %s...\n", idx+1, len(whs), wh.ID().FullyQualifiedName())
-			if wh.Name != "SNOWFLAKE" && wh.CreatedOn.Before(time.Now().Add(-4*time.Hour)) {
+			if !slices.Contains(protectedWarehouses, wh.Name) && wh.CreatedOn.Before(time.Now().Add(-2*time.Hour)) {
 				log.Printf("[DEBUG] Dropping warehouse %s, created at: %s\n", wh.ID().FullyQualifiedName(), wh.CreatedOn.String())
 				if err := client.Warehouses.Drop(ctx, wh.ID(), &DropWarehouseOptions{IfExists: Bool(true)}); err != nil {
-					return fmt.Errorf("sweeping warehouse %s ended with error, err = %w", wh.ID().FullyQualifiedName(), err)
+					log.Printf("[DEBUG] Dropping warehouse %s, resulted in error %v\n", wh.ID().FullyQualifiedName(), err)
+					errs = append(errs, fmt.Errorf("sweeping warehouse %s ended with error, err = %w", wh.ID().FullyQualifiedName(), err))
 				}
 			} else {
 				log.Printf("[DEBUG] Skipping warehouse %s, created at: %s\n", wh.ID().FullyQualifiedName(), wh.CreatedOn.String())
 			}
 		}
-		return nil
+		return errors.Join(errs...)
 	}
 }
 
 func nukeDatabases(client *Client, prefix string) func() error {
+	protectedDatabases := []string{
+		"SNOWFLAKE",
+		"MFA_ENFORCEMENT_POLICY",
+	}
+
 	return func() error {
 		log.Printf("[DEBUG] Nuking databases with prefix %s\n", prefix)
 		ctx := context.Background()
 
-		dbs, err := client.Databases.Show(ctx, &ShowDatabasesOptions{Like: &Like{Pattern: String(prefix)}})
+		var like *Like = nil
+		if prefix != "" {
+			like = &Like{Pattern: String(prefix)}
+		}
```

Comment:
Hmm, this function is very similar to `nuke*`, and some of the code could be reused.

Reply:
There is a TODO `// TODO [SNOW-955520]: generalize nuke methods (sweepers too)` on `nukeWarehouses`